### ISSUE TITLE
Roadmap for making Type Handlers more strict

### ISSUE THREAD

**Issue by jdrueckert** [MovingBlocks/Terasology#5074]
### Motivation

Currently, most of our type handlers work with a "best effort" approach.
This means for instance, that if a part of a prefab cannot be deserialized, this is (at most) logged but otherwise ignored and the rest of the prefab deserialized as well as possible.
This can lead to weird in-game behavior as some components might be present while others aren't, but there's logic depending on both and not being hardened properly against some components not being present. In addition, it also complicates debugging such weirdness.

By making the serialization/deserialization logic more strict, we can hopefully catch issues earlier and debug them easier.

### Proposal

Change the type handling logic to fail when something cannot be serialized/deserialized.
The changes will be mostly done in the engine and its subsystem for type handling.
However, the impact of the changes might well affect module land as well, especially if they unmask issues we currently already have but didn't see due to the best effort approach.

The "definition of done" for this effort is every type handler being adjusted to no longer use a best effort approach but fail with an indicative error message on failure to serialize / deserialize.

### Concerns

None so far that come to my mind, but I'm happy about any comments on aspects I might be overlooking at the moment.

_* Is there specific expertise that will be needed for this effort?_
_* Does this effort have dependencies on other efforts?_
_* Do you expect this effort to conflict with any other efforts?_
_* What are potential drawbacks of the effort?_
_* What are maintenance or continuous efforts that will persist beyond the completion of this effort?_

### Task Breakdown

- [ ] List all relevant type handlers that currently use a best effort approach
- [ ] Parallelizable: Change the type handler implementation to fail strictly
- [ ] Ideally: Increase test coverage for type handlers

_* What are the individual tasks that need to be done to complete the effort?_
_* Can you roughly estimate how hard the individual tasks would be for a software developer with 2 years of on-the-job Java development expertise, but no in-depth expertise in special areas such as rendering or AI?_
_* Which tasks are inter-dependent?_
_* Which tasks can be done in parallel?_

### Additional notes

_Feel free to add any other context or screenshots about your roadmap draft here._

**Comment by soloturn** [comment]
> **AI-assisted update.** Filed by agent driven by @soloturn via [GDD](https://github.com/soloturn/yggdrasil).

## First checklist item: list all relevant type handlers

63 `TypeHandler` implementations exist across what's checked out locally (`engine` + `subsystems/TypeHandlerLibrary` + the `Health` module - other modules aren't fetched in a base clone, so this list is necessarily partial per-repo, same caveat this issue already calls out for "module land").

One thing worth flagging up front: the base `TypeHandler.deserialize()` contract itself returns `Optional<T>`, documented as `"Optional.empty() if the value could not be deserialized"` - so returning-empty-on-bad-input isn't a lapse in individual handlers, it's the interface every one of them was written against. The real scope of this roadmap isn't fixing a handful of sloppy handlers, it's replacing a base-contract design decision that all 63 correctly implement.

### Composite/orchestrator handlers
Highest-value targets - these decide whether one bad field silently vanishes from its parent object or brings the whole thing down.

- `coreTypes/ObjectFieldMapTypeHandler.java` - the component-field deserializer every `Component` goes through. 10 catch/log-and-continue sites; central to this roadmap.
- `coreTypes/CollectionTypeHandler.java`, `ArrayTypeHandler.java`, `GenericMapTypeHandler.java`, `StringMapTypeHandler.java` - same "skip the bad element, keep the rest" pattern for `List`/`Set`/arrays/`Map`.
- `coreTypes/RuntimeDelegatingTypeHandler.java`, `FutureTypeHandler.java` - dispatch/deferred wrappers.
- `Serializer.java` - the one file in this whole tree that actually throws (`DeserializationException`), for schema-version mismatches.

### Primitive leaves
Low risk, near-identical boilerplate, `Optional.empty()` on the wrong `PersistedData` kind: `IntTypeHandler`, `LongTypeHandler`, `FloatTypeHandler`, `DoubleTypeHandler`, `ByteTypeHandler`, `ByteArrayTypeHandler`, `CharacterTypeHandler`, `BooleanTypeHandler`, `StringTypeHandler`, `NumberTypeHandler`, `EnumTypeHandler` (11).

### `StringRepresentationTypeHandler`-based domain handlers
Parse-a-string-or-give-up, higher real risk since these carry asset/URI/component-class references that go stale across engine/module versions (this is exactly the family #5359/`962df32fb` had to patch for a real crash): `SimpleUriTypeHandler`, `ComponentClassTypeHandler`, `BlockTypeHandler`, `BlockUriTypeHandler`, `BlockFamilyTypeHandler`, `NameTypeHandler`, `CollisionGroupTypeHandler`, `AssetTypeHandler`, `TextureRegionAssetTypeHandler`, `TextureRegionTypeHandler`, `UITextureRegionTypeHandler`, `UISkinTypeHandler`, `IntegerRangeHandler`, `LocaleTypeHandler` (13, plus the base class itself).

### Math/geometry
20 near-identical handlers (`Vector2f/2i/2fc/2ic`, `Vector3f/3i/3fc/3ic`, `Vector4f/4i/4fc/4ic`, `Quaternionf/fc`, `AABBf/i`, `Rectanglef/i`, `BlockArea/BlockAreac`, `BlockRegion`) - mechanical, same pattern throughout, lowest individual risk but highest count.

### Extension/domain-specific
`EntityRefTypeHandler` (touched by #5365, currently in review), `NetEntityRefTypeHandler`, `ColorTypeHandler`/`ColorcTypeHandler`, `ChunkMeshTypeHandler`, `ModifiableValueTypeHandler`, `AutoConfigTypeHandler`.

### Reflection/type-resolution layer
Controls whether a class reference resolves at all, upstream of everything else: `ModuleEnvironmentSandbox`, `ReflectionsSandbox`, `SerializationSandbox`.

### Module-level
`Health` is the only module fetched locally: `InstantTypeHandler`, `DurationTypeHandler`.

## A caution on the second checklist item

Recent, unrelated work in this exact area went the other way. `962df32fb` (this week) fixed a real crash - a saved player entity referencing a renamed/removed component class NPE'd on load - by making `SimpleUri`/`StringRepresentationTypeHandler` *more* lenient (skip the malformed entry, keep loading) rather than failing loudly. Converting these handlers to fail-strict without also solving save/version-compatibility tolerance would trade "silent weirdness" for "won't load your old save" - which is worse for players in the common case of a renamed/removed component after an engine or module update. Worth deciding how that tension gets resolved before starting on the composite handlers above, since they're exactly where it'll bite first.
